### ISSUE TITLE
Calling \Browser::isMobile() throws the error: "Argument 1 passed to UAParser\Parser::parse() must be of the type string, null given"

### ISSUE THREAD

**Issue by paulcdejean** [hisorange/browser-detect#153]
Hello after I updated this package I'm getting this error:

```
Argument 1 passed to UAParser\Parser::parse() must be of the type string, null given, called in /sites/152577226/vendor/hisorange/browser-detect/src/Stages/UAParser.php on line 25
```

The code causing this is simply:

`if (\Browser::isMobile()) {`

It seems like in the update you tried to add some stronger typing, but there was some error.

The package update that resulted in this error was updating from 3.12.4 to 3.12.5

**Comment by hisorange** [comment]
I will look into it!

**Comment by paulcdejean** [comment]
Thanks. If you Google this error you'll see a lot of people are seeing it
right now.

**Comment by hisorange** [comment]
Could You please send me a composer.lock to my private email address? I would like to reproduce the exact versions for the test, because the unit test did not catch it at the time, and it seems like the UAParser added typehints as well, and I may need to release a version lock retroactively.

Email address for this issue: browser.error@hisorange.me

**Comment by hisorange** [comment]
https://github.com/ua-parser/uap-php/commit/5afbd8505c0791c286c1a75d8d2ffbd506b3ec3f#diff-1041b02aae148a89853dc0d2710afc9c

I found the source of the issue, this release was a patch level release but it broke everyone's code who relied on them, I will have to do some more research. But from the commit and the version number it seems like the have released a breaking change in a patch. 

To solve this I will release a backport for 3.x and lock their version before the typehint was introduced.

Thank You, for raising attention, will make a backport today.

If the problem is live in production for You, lock the uaparser's version to 3.9.7 in your composer.json this will solve the issue for now.

**Comment by matthijs-neijenhuijs** [comment]
will there be a new release?

**Comment by paulcdejean** [comment]
Thanks, I'm testing locking the uaparser version to 3.9.7 on our testing environment now.

It looks like the change they made was very irresponsible to make as a patch version change, no one's perfect though.

**Comment by hisorange** [comment]
Sorry guys, I would love to release a new version lock to every version. But the Q3 of the year is starting and I am over crowded with job around this time. I know they made a mistake, it happens and I know they can fix it as well (simply release a revert commit as a new minor) but I will not going to be able to path this library until the next week or so.

For quick fix, a composer.json change will help the issue for everyone.

**Comment by hisorange** [comment]
Released a backport as 3.1.6 this will keep the 3.x versions safe as well.

Thanks for the report, have a nice day!
